GSoC 2017 Proposal - Data Visualization For Archived Content

Blackglade opened the pull request:
@b5 Here is the proposal for my idea that we discussed on slack!

Relevant other people I should probably tag: @dcwalk @titaniumbones 

Also, incase you don't want to read this markdown file, I also have an adjacent Google doc with the write up available as well! Find it here: https://docs.google.com/document/d/11iNZW0QVtRH8JasixzObAs7pMLOuOH9V3NJJUY-WPJM/edit?usp=sharing
Let me know what you think! 

***Note: I purposely left it a bit vague mostly because I wanted some feedback before I chose a definite model/graph to work on!***

Blackglade committed:
My Proposal

dcwalk commented:
Thanks for sharing @Blackglade! I'll be able to take a look at this tonight.

b5 commented on gsoc/proposal-Data-Visualization-For-Archived-Content.md:
```diff
@@ -0,0 +1,97 @@
+This data, while vital for the community as a whole, is useless in the hands of the average individual. To put it in simple terms, the average laymen just doesn’t have the time or the enormous effort required to sit through each changing variable and try to decipher which of it is relatively “meaningful.” Most individuals looking to do research/reports in these areas, although happy to discover t ...
+
+**Describe how your project meets this need:**  
+This is where my project comes in. Using data visualization libraries (such as https://d3js.org/), my project’s goal will be to develop a set of models/graphs accurately displaying any “meaningful” data you have archived. I would ideally be using the DataRefuge API for this project as a means to gather data for my in-browser model, but could eventually directly integrate with your toolkit to make ...
```
If accepted we would expect a fairly tight integration between react & d3 for performance reasons for end users. As a suggested workflow you could first build the graphs in pure d3js, and we'd then guide you through the steps for integrating with react's rendering engine, using keeping the underlying d3-core layout functions & your d3-oriented code. We would make sure you're fully supported during this process.

b5 commented on gsoc/proposal-Data-Visualization-For-Archived-Content.md:
```diff
@@ -0,0 +1,97 @@
+This is where my project comes in. Using data visualization libraries (such as https://d3js.org/), my project’s goal will be to develop a set of models/graphs accurately displaying any “meaningful” data you have archived. I would ideally be using the DataRefuge API for this project as a means to gather data for my in-browser model, but could eventually directly integrate with your toolkit to make ...
+
+**Milestones/Timeline:**  
+This project is essentially composed of **3 major milestones**: working with my mentors to identify which data is meaningful and worth developing a model for, developing the model while simultaneously receiving feedback on improvements, and then eventually committing it to production for use. Below you will find a rough timeline:
```
These milestones seem great, the only change I'd advise is we'd ask ask you to commit continuously to production, at first with a minimum viable version of an initial visualization, and then continuously improve it over time before moving on to the next. Your third milestone could instead be documenting what you learned for the greater community to read about.

b5 commented on gsoc/proposal-Data-Visualization-For-Archived-Content.md:
```diff
@@ -0,0 +1,97 @@
+
+
+**Deliverables:**  
+A set of models/graphs for visualization archived data (I say a set instead of a specific number, because I am not too sure going in what data we would want to display. Realistically I would like to work on from anywhere between 3 to 5 models). 
```
This is great! The only thing I'd add is a final recap document on what you'd learned and how the community came to benefit from your work (as mentioned above)

b5 commented on gsoc/proposal-Data-Visualization-For-Archived-Content.md:
```diff
@@ -0,0 +1,97 @@
+**Commitment:**
+In regards to commitment, I will try to dedicate about 30 hours of my week focused on this task. I have no other commitments for this summer so far, nor any vacation plans. I will be able to fully focus on this project.
+
+### Thank you for taking the time to reading my proposal and I hope you will consider me for GSoC 2017!
```
Thank you for submitting!

b5 reviewed (commented):
This is great Harsh! I left a few small comments that you may wish to use to alter your proposal, but the general outline of what you've proposed here looks great to me. Thanks for submitting!

suchthis commented on gsoc/proposal-Data-Visualization-For-Archived-Content.md:
```diff
@@ -0,0 +1,97 @@
+Finalize any ideas I had and hopefully commit my developed models to the “live” website 
+
+* **Aug 8th - Aug 22nd**
+Reach out to individuals using the website as well as my mentors to see if I can identify any other changes to further optimize the experience and make it more interactive and meaningful on the end-user. As with any product being shipped, testing usage and interaction with the product is important to gauge its overall perception. 
```
Thanks for this thoughtful inclusion! I love that you've planned ahead for a round of post-deploy refinement based on user feedback.

mhucka replied:
I also appreciate the explanations of what is expected during each time period. All too often, proposals have only a bullet list and leaves too much to the imagination.

mhucka reviewed (commented):
Minor comments

Blackglade committed:
Refined Proposal

Blackglade commented:
Thank you @mhucka @suchthis for your comments 😄 !

@b5  The initial plan was to finish one model, before moving onto the next one, but I wasn't too sure how to properly write that into a timeline format split up into distinctive sections. Anyway, I've updated the proposal to make sure I specified that I will be doing exactly that. I've also changed my milestones so I'm committing continuously and have a part of the timeline dedicated to documenting what I learned. The only parts that really changed is the **Describe how your project meets this need**, **Milestones/Timeline** , and **Deliverables** section.

Thanks again for your feedback and let me know if there is anything else I could change before submitting it on the website (don't really want to wait until the last day to get this in). Thanks again for your help.

Oh and as always, if you don't want the markdown file, there is an updated version on the google doc link: https://docs.google.com/document/d/11iNZW0QVtRH8JasixzObAs7pMLOuOH9V3NJJUY-WPJM/edit

suchthis commented:
Thanks for submitting an update to your proposal, @Blackglade! It's really nice to see you incorporating the feedback to include continuous integration and a documentation step. @b5 will have a better perspective on whether to further update your timeline, e.g. whether your allotment of time for information gathering vs. model building seems accurate. If you would still like to build into the proposal the idea that you'll be working on models sequentially, you could just add that you'll be defining a set of prioritized models, and then working during the model-delivery period on each model in sequence, committing a model to production when complete, and working on the highest-priority views first.

But overall I think this is a great proposal!

dcwalk commented:
Thank you for submitting a proposal for Google Summer of Code! 🎉 
I am merging this in as we move into the review phase. Based on [Google's timeline](https://developers.google.com/open-source/gsoc/timeline), on May 4 accepted proposals will be announced
